modules/study/src/main/BSONHandlers.scala

```diff
@@ -137,81 +139,82 @@ object BSONHandlers:
       comments       = doc.getAsOpt[Comments](F.comments) getOrElse Comments.empty
       gamebook       = doc.getAsOpt[Gamebook](F.gamebook)
       glyphs         = doc.getAsOpt[Glyphs](F.glyphs) getOrElse Glyphs.empty
-      score          = doc.getAsOpt[Score](F.score)
+      eval           = doc.getAsOpt[Score](F.score).map(_.eval)
       clock          = doc.getAsOpt[Centis](F.clock)
-      crazy          = doc.getAsOpt[Crazyhouse.Data](F.crazy)
+      crazyData      = doc.getAsOpt[Crazyhouse.Data](F.crazy)
       forceVariation = ~doc.getAsOpt[Boolean](F.forceVariation)
-    } yield Node(
-      id,
-      ply,
-      Uci.WithSan(uci, san),
-      fen,
-      check,
-      shapes,
-      comments,
-      gamebook,
-      glyphs,
-      score,
-      clock,
-      crazy,
-      Node.emptyChildren,
-      forceVariation
+    } yield Branch(
+      id = id,
+      ply = ply,
+      move = Uci.WithSan(uci, san),
+      fen = fen,
+      check = check,
+      shapes = shapes,
+      comments = comments,
+      gamebook = gamebook,
+      glyphs = glyphs,
+      eval = eval,
+      clock = clock,
+      crazyData = crazyData,
+      children = Branches.empty,
+      // TODO we probably need to have access to the variant to set the opening correctly
+      forceVariation = forceVariation
     )
 
-  private[study] def writeNode(n: Node) =
-    import Node.BsonFields.*
+  private[study] def writeBranch(n: Branch) =
+    import Node.{ BsonFields as F }
     val w = new Writer
     $doc(
-      ply            -> n.ply,
-      uci            -> n.move.uci,
-      san            -> n.move.san,
-      fen            -> n.fen,
-      check          -> w.yesnoO(n.check),
-      shapes         -> n.shapes.value.nonEmpty.option(n.shapes),
-      comments       -> n.comments.value.nonEmpty.option(n.comments),
-      gamebook       -> n.gamebook,
-      glyphs         -> n.glyphs.nonEmpty,
-      score          -> n.score,
-      clock          -> n.clock,
-      crazy          -> n.crazyData,
-      forceVariation -> w.boolO(n.forceVariation),
-      order -> {
+      F.ply            -> n.ply,
+      F.uci            -> n.move.uci,
+      F.san            -> n.move.san,
+      F.fen            -> n.fen,
+      F.check          -> w.yesnoO(n.check),
+      F.shapes         -> n.shapes.value.nonEmpty.option(n.shapes),
+      F.comments       -> n.comments.value.nonEmpty.option(n.comments),
+      F.gamebook       -> n.gamebook,
+      F.glyphs         -> n.glyphs.nonEmpty,
+      F.score          -> n.eval.flatMap(_.score), // BC stored as score (maybe its better to keep this way?)
```

Comment:
After thinking about it, I think keeping only the score is fine, the best move is I guess only used to display (best move was XX) in game analysis.